src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -69,7 +70,8 @@ internal UnitTestDiscoverer()
                     "MSTestDiscoverer: Warning during discovery from {0}. {1} ",
                     source,
                     warning);
-                logger.SendMessage(TestMessageLevel.Warning, warning);
+                var message = string.Format(CultureInfo.CurrentCulture, Resource.DiscoveryWarning, source, warning);
```

Comment:
Could we do the same for execution too please? :)

Reply:
will consider in separate PR.

Reply:
sure. Please go ahead.